---
title: C26110
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C26110
helpviewer_keywords:
- C26110
ms.assetid: d82b79ec-6d7f-438b-bd6a-da874a3e08e5
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: 902455e15e52a36a682197c3c265aa97f53a5161
ms.sourcegitcommit: 94b3a052fb1229c7e7f8804b09c1d403385c7630
ms.translationtype: MT
ms.contentlocale: ko-KR
ms.lasthandoff: 04/23/2019
ms.locfileid: "62428133"
---
# <a name="c26110"></a>C26110
warning C26110: 호출자가 잠금 유지 실패 \<잠금 > 함수를 호출 하기 전에 \<func >.

 잠금이 필요한 경우 함수 자체 또는 해당 호출자가 잠금을 획득해야 하는지 여부를 확인합니다. `_Requires_lock_held_` 주석 위반이 있으면 경고 C26110이 발생합니다.

## <a name="example"></a>예제
 다음 예제에서는 `_Requires_lock_held_` 함수의 `LockRequired` 주석에 `LockRequired` 호출자가 `LockRequired`를 호출하기 전에 잠금을 획득하도록 지정되었기 때문에 경고 C26110이 생성되었습니다. 이 주석이 없으면 `LockRequired`가 잠금으로 보호되는 공유 데이터에 액세스하기 전에 잠금을 획득해야 합니다.

```

typedef struct _DATA
{
    CRITICAL_SECTION cs;
    int d;
} DATA;

_Requires_lock_held_(p->cs)

void LockRequired(DATA* p)
{
    p->d = 0;
}

void LockNotHeld(DATA* p)
{
    LockRequired(p); // Warning C26110
}
```